spyder/api/widgets.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+spyder.api.widgets
+==================
+
+This package exposes shareable widgets that can be used in
+third-party plugins.
+"""
+
+from spyder.widgets.tabs import Tabs
+from spyder.widgets.comboboxes import PythonModulesComboBox
+from spyder.widgets.variableexplorer.texteditor import TextEditor
```

Comment:
Just for context, in the plugins that I am familiar with (line_profiler, memory_profiler, unittest) `TextEditor` is used to show a block of text to the user. The edit functionality is not used at all. In the long run, it may be possible to use a function which is not so deeply coupled.